hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/util/UserAgentParser.java

```diff
@@ -30,15 +31,21 @@ public class UserAgentParser {
   public UserAgentParser(Config config) {
     Config cacheConfig = config.getConfig(CACHE_CONFIG_KEY);
     this.userAgentMaxLength = config.getInt(USER_AGENT_MAX_LENGTH_KEY);
+    int cacheMaxSize = cacheConfig.getInt(CACHE_CONFIG_MAX_SIZE);
     this.userAgentCache =
         CacheBuilder.newBuilder()
-            .maximumSize(cacheConfig.getInt(CACHE_CONFIG_MAX_SIZE))
+            .maximumSize(cacheMaxSize)
             .expireAfterAccess(cacheConfig.getDuration(CACHE_CONFIG_ACCESS_EXPIRATION_DURATION))
             .recordStats()
             .build(CacheLoader.from(userAgentStringParser::parse));
+    registerCacheMetrics("userAgentCache", userAgentCache, cacheMaxSize);
+  }
 
-    PlatformMetricsRegistry.registerCache(
-        this.getClass().getName() + ".userAgentCache", userAgentCache, Collections.emptyMap());
+  private void registerCacheMetrics(String cacheNameSuffix, Cache cache, int cacheMaxSize) {
+    String cacheName = this.getClass().getName() + "." + cacheNameSuffix;
+    PlatformMetricsRegistry.registerCache(cacheName, cache, Collections.emptyMap());
```

Comment:
We don't need `PlatformMetricsRegistry.registerCache(cacheName, cache, Collections.emptyMap());` right @kotharironak ? 

Thanks for pointing it out @sanket-mundra

Reply:
Ah! I thought `registerCacheTrackingOccupancy` register only `cache max size`. Hmm, the name could have been `registerCacheAndTrackingOccupancy`.

Reply:
Ok, re-creating the PR.